Catalog

de-vasta opened the pull request.

de-vasta committed:
✨ feat(styles): add product catalog layout with basic styling

- Introduce variables for colors, sizes, and fonts

de-vasta committed:
✨ feat(catalog): implement product catalog layout

- implement responsive grid layout for product cards
- add hover effect to product cards
- use variables for colors and transition times
- fix: update demo and test report links in readme
- fix: update checklist in readme

mateacademy-ai-mentor commented on .linthtmlrc.json:
```diff
@@ -26,24 +26,15 @@
-    "font",
-    "s"
-  ],
+  "tag-bans": ["b", "i", "u", "center", "style", "marquee", "font", "s"],
```
The list of banned tags is incomplete. According to the code style rules, tags like `<strike>`, `<tt>`, `<big>`, and `<small>` should also be banned to enforce the use of modern semantic HTML and CSS alternatives.

mateacademy-ai-mentor commented on .linthtmlrc.json:
```diff
@@ -26,24 +26,15 @@
   "title-no-dup": true,
   "line-end-style": "lf",
-  "attr-new-line": 2,
+  "attr-new-line": false,
```
This setting contradicts the code style rules, which state: "Put attributes on a new line for better readability if there are more than 3 attributes". To enforce this, you should set this property to `3` instead of `false`.

mateacademy-ai-mentor commented on src/styles/header.scss:
```diff
@@ -0,0 +1,68 @@
+@use 'variables.scss' as *;
+
+.header {
+  background-color: white;
```
It's a good practice to use the existing `$card-bg-color` variable (which is `#fff`) here instead of the keyword `white` for consistency across the project. This aligns with the requirement to use variables for colors.

mateacademy-ai-mentor commented on src/styles/header.scss:
```diff
@@ -0,0 +1,68 @@
+  align-items: center;
+  z-index: 1;
+
+  box-shadow: 0 2px 4px 0 #0000000d;
```
This violates checklist item #2: 'Repeated sizes and special colors are put to variables'. The color `#0000000d` should be stored in a variable to avoid using "magic numbers" in the code. This makes the code more readable and easier to maintain.

mateacademy-ai-mentor commented on src/styles/index.scss:
```diff
@@ -1,3 +1,9 @@
+:root {
+  font-family: Roboto, sans-serif;
+  font-weight: 500;
+  font-size: 12px;
```
According to the code style rules, the root `font-size` should not be changed from the browser's default (usually 16px). Please apply these base font styles to the `body` selector instead of `:root`. This violates checklist item #6: 'Code follows all the Code Style Rules ❗️'.

mateacademy-ai-mentor commented on src/index.html:
```diff
@@ -2,26 +2,265 @@
+          <span class="product-card__price-value">$2,199</span>
+        </div>
+        <a href="#" class="product-card__btn" data-qa="card-hover">buy</a>
+      </article>
```
According to the task description, the `data-qa="card"` and `data-qa="card-hover"` attributes should only be applied to the *first* product card and its corresponding 'buy' button. Please remove these attributes from all subsequent cards.

mateacademy-ai-mentor commented on src/styles/product-card.scss:
```diff
@@ -0,0 +1,115 @@
+    text-transform: uppercase;
+    text-decoration: none;
+
+    border: none;
```
This violates checklist item #5: 'All changes on `:hover` are smooth'. When a border is added on `:hover` but doesn't exist in the default state, it causes a slight layout shift as the element's size changes. To prevent this jump and ensure a truly smooth transition, it's better to declare a transparent border of the same width in the default state. This reserves the space for the border without making it visible.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great job on the overall structure and SCSS implementation! Your responsive grid is working well. However, I'm requesting a few changes to align the solution with the task requirements.

The most critical issue is in `src/index.html`. The `data-qa="card"` and `data-qa="card-hover"` attributes are required **only** on the first product card and its corresponding 'Buy' button, but they have been applied to all of them. 

Additionally, please address the following to meet the code style requirements:
- In `src/styles/index.scss`, move the base font styles from the `:root` selector to the `body` selector as per the project's style guide. [CHECKLIST ITEM #6]
- In `src/styles/product-card.scss`, add a transparent border to the button's default state to prevent the layout from shifting on hover. [CHECKLIST ITEM #5]

Once these points are addressed, your work will be ready for approval. Keep up the excellent effort!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

de-vasta committed:
🐛 fix(linthtml): update linting rules and html structure

- add more tag bans to `.linthtmlrc.json`
- remove data-qa attributes from `index.html`
- change background-color for header
- change body style to body tag
- add border to buy button
- change header border color variable

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: NCM]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.
